lib/core/proxy.c

```diff
@@ -371,6 +371,17 @@ static int on_body(h2o_http1client_t *client, const char *errstr)
     return 0;
 }
 
+static char compress_hint_to_enum(const char *val, size_t len)
+{
+    if (!strncasecmp("ON", val, len)) {
```

Comment:
Please use `h2o_lcstris` instead of `strncasecmp` so that the comparison would be a complete match (instead of a prefix match; current code would yield true if [val,len] is ["O", 1]) and that it would be locale-independent.